### PR TITLE
Support POSIX environment for build boss

### PR DESCRIPTION
**Pull request by pazof**

### CODE REVIEW

File: src/Tools/BuildBoss/Program.cs [hunk 1]
```diff
@@ -1,4 +1,5 @@
-﻿// Licensed to the .NET Foundation under one or more agreements.
+﻿using System.Net;
```
Comment (333fred) [inline src/Tools/BuildBoss/Program.cs:1]: This needs to be with the rest of the usings please.

**Review by 333fred — COMMENTED** [review]
@pazof can you please update the pr description to include what this change is and why it's necessary?

**Comment by pazof** [comment]
I'm open to any proposal, for the title :-)

**Comment by 333fred** [comment]
My suggestion would be "Support POSIX paths in build boss"

**Comment by RikkiGibson** [comment]
Could you please describe the scenario in which this change has a positive effect? Perhaps it enables running certain Roslyn scripts from Linux/Mac?

**Comment by pazof** [comment]
This Build Boss won't start without this kind of change, in any kind of installation hosted by a POSIX file system, 
With this change, it starts, do its works and ends without error.

**Comment by pazof** [comment]
I cannot understand how the Path.Combine(Path.Combine(string, string), string) gave no error from the CI, 
while Path.Combine(string,string,string) breaks all

File: src/Tools/BuildBoss/Program.cs [hunk 2]
```diff
@@ -138,14 +138,14 @@ private static bool ProcessSolution(string solutionFilePath, bool isPrimarySolut
...
-            var targetsDirectory = Path.Combine(repositoryDirectory, @"eng\targets");
+            var targetsDirectory = Path.Combine(repositoryDirectory, Path.Combine("eng", "targets"));
             var checker = new TargetsCheckerUtil(targetsDirectory);
             return CheckCore(checker, $"Targets {targetsDirectory}");
         }
 
         private static bool ProcessStructuredLog(string artifactsDirectory, string configuration)
         {
-            var logFilePath = Path.Combine(artifactsDirectory, $@"log\{configuration}\Build.binlog");
+            var logFilePath = Path.Combine(Path.Combine(Path.Combine(artifactsDirectory, "log"), "{configuration}"), "Build.binlog");
```
Comment (RikkiGibson) [inline src/Tools/BuildBoss/Program.cs:148]: It looks like `"{configuration}"` was meant to be `$"{configuration}"`.

Comment (pazof) [reply src/Tools/BuildBoss/Program.cs]: yes

Comment (RikkiGibson) [reply src/Tools/BuildBoss/Program.cs]: I think the issue wasn't resolved here--we don't want to delete parameter `configuration`. What we want is to actually use it to create the log file path. Which will happen if we change `"{configuration}"` to `$"{configuration}"`.

**Comment by pazof** [comment]
I need english practice :-D

File: src/Tools/BuildBoss/Program.cs (hunk 2 above)
Comment (pazof) [reply src/Tools/BuildBoss/Program.cs]: I now changed the code to use the source string, I think It's the best form.

**Comment by pazof** [comment]
I give up.

**Pull request closed without merging**